crates/ty_python_semantic/src/semantic_index/builder.rs

```diff
@@ -1875,11 +1879,50 @@ impl<'ast> Visitor<'ast> for SemanticIndexBuilder<'_, 'ast> {
                         });
                     }
                     let scope_id = self.current_scope();
+                    if let Some(nonlocals) = self.nonlocals_by_scope.get(&scope_id)
+                        && nonlocals.contains(&symbol_id)
+                    {
+                        // TODO: This should be some sort of error?
```

Comment:
I'm not sure how to report an error/diagnostic here, but once I figure that out (the same for its partner in the `Stmt::Nonlocal` branch immediately below) that should make the `error:` assert in `global.md` start passing.

Reply:
e3fc88b94af80f5a93157384508c5ec06376c945